src/zimcheck/checks.cpp

```diff
@@ -487,14 +487,19 @@ void ArticleChecker::detect_redundant_articles()
         progress.report();
         auto l = it.second;
         while ( !l.empty() ) {
-            const auto e1 = archive.getEntryByPath(l.front());
+            const auto [e1_idx, e1_cluster_idx, e1_blob_idx] = l.front();
             l.pop_front();
+            const auto e1 = archive.getEntryByPath(e1_idx);
             if ( !l.empty() ) {
                 // The way we have constructed `l`, e1 MUST BE an item
                 const std::string s1 = e1.getItem().getData();
```

Comment:
Before aliases were introduced, this extra access to the item data happened only for the hopefully rare false positives due to hash conflicts. Now the item data will be loaded during the redundant item check for each entry having an alias. If aliases are going to be used mainly for large binary content (e.g. youtube videos) then we better have this access optimized.

Reply:
Indeed. Done.